loader/src/main/java/net/neoforged/fml/javafmlmod/AutomaticEventSubscriber.java

```diff
@@ -42,22 +48,47 @@ public static void inject(final ModContainer mod, final ModFileScanData scanData
         Map<String, String> modids = scanData.getAnnotations().stream().filter(annotationData -> MOD_TYPE.equals(annotationData.annotationType())).collect(Collectors.toMap(a -> a.clazz().getClassName(), a -> (String) a.annotationData().get("value")));
 
         ebsTargets.forEach(ad -> {
-            @SuppressWarnings("unchecked")
             final EnumSet<Dist> sides = getSides(ad.annotationData().get("value"));
             final String modId = (String) ad.annotationData().getOrDefault("modid", modids.getOrDefault(ad.clazz().getClassName(), mod.getModId()));
-            final ModAnnotation.EnumHolder busTargetHolder = (ModAnnotation.EnumHolder) ad.annotationData().getOrDefault("bus", new ModAnnotation.EnumHolder(null, EventBusSubscriber.Bus.GAME.name()));
+            final ModAnnotation.EnumHolder busTargetHolder = (ModAnnotation.EnumHolder) ad.annotationData().getOrDefault("bus", new ModAnnotation.EnumHolder(null, EventBusSubscriber.Bus.AUTOMATIC.name()));
             final EventBusSubscriber.Bus busTarget = EventBusSubscriber.Bus.valueOf(busTargetHolder.value());
             if (Objects.equals(mod.getModId(), modId) && sides.contains(FMLEnvironment.dist)) {
                 try {
+                    var clazz = Class.forName(ad.clazz().getClassName(), true, layer.getClassLoader());
+
                     IEventBus bus = switch (busTarget) {
-                        case GAME -> Bindings.getGameBus();
+                        case GAME -> FMLLoader.getBindings().getGameBus();
                         case MOD -> mod.getEventBus();
+                        case AUTOMATIC -> {
+                            boolean hasGame = false, hasMod = false;
+                            for (var method : clazz.getDeclaredMethods()) {
+                                if (!Modifier.isStatic(method.getModifiers()) || !method.isAnnotationPresent(SubscribeEvent.class) || method.getParameterTypes().length != 1) {
+                                    continue;
+                                }
+
+                                var type = method.getParameterTypes()[0];
+                                if (IModBusEvent.class.isAssignableFrom(type)) {
+                                    hasMod = true;
+                                } else if (Event.class.isAssignableFrom(type)) {
+                                    hasGame = true;
+                                }
+                            }
+
+                            if (hasGame && hasMod) {
```

Comment:
Although I understand the idea, but why do we limit this to one bus per listener?

Reply:
Oke perfect, was just curious.  Thanks for the explanation. Nice work!

Reply:
> This approach does not provide meaningful separation (there is no reason that these must be separated at the `class` level, aside from the current EBS implementation), and unifying these events in the same class can readily be achieved via `IEventBus#addListener`.

Despite the explained "limitation", leaving a class level separation here does not fix #140's main expectations.

Moving further to change the EBS implementation would be better than an incomplete solution imo.

Reply:
There is no raw variant of `addListener` that would support that use case, and reflection based consumers on the neo side is not acceptable, both because of performance concerns and because it will make unregistering those listeners impossible. In any case, I've decided to not proceed with this PR further because more magic will increase confusion. 

Instead, we should make the bus parameter required and the mod id default to first in the file (instead of all).